src/HttpParser.jl

```diff
@@ -85,6 +85,34 @@ type ParserSettings
     on_message_complete_cb::Ptr{Void}
     on_chunk_header::Ptr{Void}
     on_chunk_complete::Ptr{Void}
+
+    function ParserSettings(
+      on_message_begin_cb::Ptr{Void} = C_NULL,
+      on_url_cb::Ptr{Void} = C_NULL,
+      on_status_complete_cb::Ptr{Void} = C_NULL,
+      on_header_field_cb::Ptr{Void} = C_NULL,
+      on_header_value_cb::Ptr{Void} = C_NULL,
+      on_headers_complete_cb::Ptr{Void} = C_NULL,
+      on_body_cb::Ptr{Void} = C_NULL,
+      on_message_complete_cb::Ptr{Void} = C_NULL,
+      on_chunk_header::Ptr{Void} = C_NULL,
+      on_chunk_complete::Ptr{Void} = C_NULL)
+        function dummy(parser)
+          0
+        end
+        dummy_cb = cfunction(dummy, HTTP_CB...)
+        new(
+          on_message_begin_cb == C_NULL ? dummy_cb : on_message_begin_cb,
+          on_url_cb == C_NULL ? dummy_cb : on_url_cb,
```

Comment:
And I believe `http_cb` and `http_data_cb` need two different dummies (or two methods of it at least).